src/core/lib/gprpp/time.h

```diff
@@ -26,20 +26,27 @@
 #include "absl/types/optional.h"
 
 #include <grpc/event_engine/event_engine.h>
-#include <grpc/support/log.h>
 #include <grpc/support/time.h>
 
 #include "src/core/lib/gpr/time_precise.h"
 #include "src/core/lib/gpr/useful.h"
 
-#define GRPC_LOG_EVERY_N_SEC(n, format, ...)                    \
+#define GRPC_LOG_EVERY_N_SEC(n, severity, format, ...)          \
   do {                                                          \
     static std::atomic<uint64_t> prev{0};                       \
+    static std::atomic<bool> first_call{true};                  \
     uint64_t now = grpc_core::Timestamp::FromTimespecRoundDown( \
                        gpr_now(GPR_CLOCK_MONOTONIC))            \
                        .milliseconds_after_process_epoch();     \
+    bool should_log = false;                                    \
```

Comment:
Instead of another atomic, could we do 

```
uint64_t prev_tsamp = prev.exchange(now)
if (prev_tsamp == 0 || now - prev_tsamp > (n)*1000) {
  gpr_log(severity, format, __VA_ARGS__);
}

```

Reply:
There's a minor oddity with that approach which is that it could be called multiple times within the first millisecond of the process.

WDYT?

Reply:
Thinking about this more, I followed your suggestion. The simpler code is probably worth this edge case being mis-handled (and it could only cause an increase of logs at the very start of a process).